demo/site/src/middleware/domainRewrite.ts

```diff
@@ -5,6 +5,12 @@ import { CustomMiddleware } from "./chain";
 
 export function withDomainRewriteMiddleware(middleware: CustomMiddleware) {
     return async (request: NextRequest) => {
+        // These paths don't require a rewrite
+        const pathname = request.nextUrl.pathname;
+        if (pathname.startsWith("/block-preview/") || pathname === "/robots.txt" || pathname.startsWith("/api/")) {
```

Comment:
In the App Router Route Handlers (i.e. API Routes) can be anywhere in the page. So having /api here is a convention. Should we still keep it?

Reply:
You might use the domain param inside a api route by placing the api route into `[domain]` (and want the domain rewrite for api route too)

Reply:
So we define the following:

1. API routes in `src/app/api/` shouldn't be handled by the middleware
2. API routes in `src/app/[domain]/` should be handled by the middlware

What's the use case for "global" API routes? The site preview?

Reply:
> What do you mean by the first point. Actually they are handled in the middleware in this very PR, they're just not rewritten though.

You're right, my bad.